File: src/bun.js/bindings/JSRuntimeType.zig
/// Represents JavaScript runtime value types
pub const JSRuntimeType = enum(u16) {
    Nothing = 0x0,
    Function = 0x1,
    Undefined = 0x2,
    Null = 0x4,
    Boolean = 0x8,
    AnyInt = 0x10,
    Number = 0x20,
    String = 0x40,
    Object = 0x80,
    Symbol = 0x100,
    BigInt = 0x200,

    _,
};

File: src/jsc/JSRuntimeType.rs
/// Represents JavaScript runtime value types
// Any u16 bit-pattern is a valid value (the discriminants are bitflags). A
// `#[repr(u16)] enum` would be UB for unlisted values, so this is a transparent
// newtype with associated consts instead.
#[repr(transparent)]
#[derive(Copy, Clone, Eq, PartialEq, Hash)]
pub struct JSRuntimeType(pub u16);

impl JSRuntimeType {
    pub const NOTHING: Self = Self(0x0);
    pub const UNDEFINED: Self = Self(0x2);
    pub const NULL: Self = Self(0x4);
}
